### PR TITLE
Fix the ElasticaDatagridBuilder when use autocomplete filter

### LINKED ISSUE #24: Notice: Undefined offset: 1
When I use "doctrine_orm_model_autocomplete" filter I get following notice:

> Notice: Undefined offset: 1
> in vendor/sonata-project/admin-search-bundle/Builder/ElasticaDatagridBuilder.php at line 137

Here is simple admin config example:

``` php
    protected function configureDatagridFilters(DatagridMapper $datagridMapper)
    {
        $datagridMapper
            ->add(
                'parent',
                'doctrine_orm_model_autocomplete',
                [
                    'show_filter' => true,
                ],
                null,
                [
                    'property'    => 'name',
                ]
            );
    }
```

and the code which is referenced by the notice above:

``` php
    // first : validate if elastica is asked in the configuration for this action
    $fullCurrentAction = $admin->getRequest()->attributes->get('_controller');
    $currentAction = explode('::', $fullCurrentAction);
    // remove Action from 'listAction'
    $currentAction = substr($currentAction[1], 0, -strlen('Action'));
```

In this case (autocomplete action), $fullCurrentAction equals to "sonata.admin.controller.admin:retrieveAutocompleteItemsAction".

Possibly relates to #22.

### PR DESCRIPTION
**Pull request by chervaliery**
Closes #24 
### Changelog

``` markdown
### Fixed
 - `ElasticaDatagridBuilder` now handles autocomplete filter
```
### Todo
- [x] separate commits
- [ ] with meaningful messages
### Subject

Fixe the ElasticaDatagridBuilder and the DependencyInjection/Configuration according to the discussion in #24

### CODE REVIEW

**Comment by greg0ire** [comment]
Thanks @chervaliery ! Can you read and apply this though : https://github.com/sonata-project/SonataAdminSearchBundle/blob/master/CONTRIBUTING.md#the-commit-message ?

File: Builder/ElasticaDatagridBuilder.php [hunk 1]
```diff
@@ -131,10 +131,10 @@ public function isSmart(AdminInterface $admin, array $values = array())
     {
 
         // first : validate if elastica is asked in the configuration for this action
-        $fullCurrentAction = $admin->getRequest()->attributes->get('_controller');
-        $currentAction = explode('::', $fullCurrentAction);
+        $logicalControllerName = $admin->getRequest()->attributes->get('_controller');
+        $currentAction = explode(':', $logicalControllerName);
```
Comment (greg0ire) [inline Builder/ElasticaDatagridBuilder.php:135]: This should be in a separate commit (same PR is fine)

File: DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -33,7 +33,7 @@ public function getConfigTreeBuilder()
                       ->children()
                           ->scalarNode('finder')->isRequired()->end()
                           ->arrayNode('actions')
-                              ->defaultValue(array('list'))
+                              ->defaultValue(array('list', 'retrieveAutocompleteItems'))
```
Comment (greg0ire) [inline DependencyInjection/Configuration.php:36]: This should also be in a separate commit

**Comment by chervaliery** [comment]
It should be good this time

**Comment by greg0ire** [comment]
It's way better like this, but the third commit message looks strange : 
What do you mean by `admin_finder_services` ?

Also, make sure you associate your excellium mail address to your account if you want your contributions to show up in your account ;)

**Comment by chervaliery** [comment]
admin_finder_services is the name of the node in the configuration tree.
I don't know how to explain it more clearly.

**Comment by core23** [comment]
Please change your PR title. I can't get what this PR is about when browsing the PR list. Same for the changelog.

**Comment by chervaliery** [comment]
I rename it

**Comment by core23** [comment]
```
### Fixed
ElasticaDatagridBuilder
DependencyInjection/Configuration
```

WHAT did you fix?

**Comment by chervaliery** [comment]
Is it better ?

**Comment by chervaliery** [comment]
Is there something else to do ?

**Comment by greg0ire** [comment]
> admin_finder_services is the name of the node in the configuration tree.
> I don't know how to explain it more clearly.

Oh ok, I did not understand that it was supposed to be part of the same sentence.

Try to keep the commit message subject < 50 characters, like I asked, and I will merge this. See how your sentence is broken in the middle [here](https://github.com/sonata-project/SonataAdminSearchBundle/pull/41/commits/0fddf6d6a08756713364533be2c3df589ca89321) ? This is what you should try to avoid.

So

```
Small message under fifty characters

Empty line followed by a § that explains in detail what you are doing.
```

**Comment by chervaliery** [comment]
I change the commit message and title

**Comment by greg0ire** [comment]
Great job! Let's wait for the build to complete!

**Comment by greg0ire** [comment]
Thanks again, nice job @chervaliery !

**Pull request merged**